src/common/helpers/snap-builds.js

```diff
@@ -25,43 +25,116 @@
 // };
 
 export const BuildStatusColours = {
+  BLUE: 'blue',
   GREEN: 'green',
   YELLOW: 'yellow',
   RED: 'red',
   GREY: 'grey'
 };
 
+const BuildAndPubState = {
+  'NEVER_BUILT':     createState('Never built', BuildStatusColours.GREY, false, 8),
+  'BUILDING_SOON':   createState('Building soon', BuildStatusColours.GREY, 'ellipsis', 7),
+  'IN_PROGRESS':     createState('In progress', BuildStatusColours.BLUE, 'spinner', 2),
+  'FAILED_TO_BUILD': createState('Failed to build', BuildStatusColours.RED, 'cross', 1),
+  'WONT_PUBLISH':    createState('Built, won\'t be published', BuildStatusColours.GREEN, 'tick_outlined', 6),
+  'PUBLISH_SOON':    createState('Built, publishing soon', BuildStatusColours.GREY, 'tick', 2),
+  'PUBLISH_NOW':     createState('Built, publishing now', BuildStatusColours.BLUE, 'tick', 3),
+  'PUBLISH_FAILED':  createState('Built, failed to publish', BuildStatusColours.RED, 'tick', 4),
+  'PUBLISHED':       createState('Built and published', BuildStatusColours.GREEN, 'tick_filled', 5)
+};
+
+function createState(message, colour, icon, priority) {
+  return {
+    message,
+    colour,
+    icon,
+    priority
+  };
+}
+
 // Based on BuildStatusConstants from LP API
 // https://git.launchpad.net/launchpad/tree/lib/lp/buildmaster/enums.py#n22
 //
-// mapping between build status from LP and red, yellow, green, grey status colours
-const BuildStatusMapping = {
-  'Needs building': BuildStatusColours.YELLOW,
-  'Successfully built': BuildStatusColours.GREEN,
-  'Failed to build': BuildStatusColours.RED,
-  'Dependency wait': BuildStatusColours.RED,
-  'Chroot problem': BuildStatusColours.RED,
-  'Build for superseded Source': BuildStatusColours.RED,
-  'Currently building': BuildStatusColours.YELLOW,
-  'Failed to upload': BuildStatusColours.RED,
-  'Uploading build': BuildStatusColours.YELLOW,
-  'Cancelling build': BuildStatusColours.RED,
-  'Cancelled build': BuildStatusColours.RED
+const LaunchpadBuildStates = {
+  'NEEDS': 'Needs building',
+  'SUCCESS': 'Successfully built',
+  'FAILED': 'Failed to build',
+  'WAIT': 'Dependency wait',
+  'CHROOT': 'Chroot problem',
+  'SUPERCEDED': 'Build for superseded Source',
+  'BUILDING': 'Currently building',
+  'UPLOAD_FAIL': 'Failed to upload',
+  'UPLOADING': 'Uploading build',
+  'CANCELLING': 'Cancelling build',
+  'CANCELLED': 'Cancelled build'
+};
+
+const LaunchpadStoreUploadStates = {
+  'UNSCHEDULED': 'Unscheduled',
+  'PENDING': 'Pending',
+  'FAILED_UPLOAD': 'Failed to upload',
+  'FAILED_RELEASE': 'Failed to release to channels',
+  'UPLOADED': 'Uploaded'
```

Comment:
[Ditto](http://bazaar.launchpad.net/+branch/launchpad/view/head:/lib/lp/snappy/interfaces/snapbuild.py#L81).  (The Git import is a bit out of date so you get a slow Bazaar link instead.)